src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +33,34 @@
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${selectTodo.id}`}
             </div>
 
             {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => setSelectTodo(null)}
+            />
           </header>
 
           <div className="modal-card-body">
             <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
+              {selectTodo.title}
             </p>
 
             <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
+              <strong
+                className={
+                  selectTodo.completed ? 'has-text-success' : 'has-text-danger'
+                }
+              >
+                {selectTodo.completed ? 'Done' : 'Planned'}
+              </strong>
               {' by '}
 
-              <a href="mailto:Sincere@april.biz">Leanne Graham</a>
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
Ensure that `user?.email` and `user?.name` are not undefined before rendering the link. Although you are using optional chaining, it might be beneficial to provide a fallback or loading state to handle cases where the user data is not yet available.